### DISCUSSION TITLE
How to use the "run script" function in actions?

### DISCUSSION THREAD

**Discussion by Wittusus** [BartoszCichecki/LenovoLegionToolkit#1214] [Q&A]
I got a python script, tried running the action and the script didn't run, tried using a bat file that would run it instead but still it doesn't work with the correct path. What do I do?

**Comment by BartoszCichecki** [comment]
You give more details about what you tried to do.

**Reply by Wittusus** [reply]
I created a new action for "When game is running" with a single action "Run", where in executable path I tried putting my python file or bat file, with their respective paths C:\XXX\file.py or .bat, but when I turn on a game or try "run now" option it doesn't run the script. Does it only accept .exe files?

**Reply by BartoszCichecki** [reply]
It's called executable for a reason. If you want to run a python script on the command line you use python to execute it, don't you?

bat files are a bit special in Windows because it will figure it out and run it even if you put the path on the "executable" line.

**Reply by Wittusus** [reply]
Yeah but bat doesn't run either, while running on my own

**Reply by BartoszCichecki** [reply]
If it doesn't run, you are doing something wrong.

**Reply by Wittusus** [reply]
Do you have any instruction on how to do so then?

**Reply by Ace-Radom** [reply]
Only a guess: I just tried to launch cmd in debug mode, but the cmd console didn't exist. You can see in task manager that cmd.exe is "running", but as a sub-process of LLT itself. It seems to have sth to do with the logic of the `LenovoLegionToolkit.Lib.System.CMD.RunAsync` function, but I'm not 100% sure because I've never written such things in C# before. Trying to figure out what's happening.

**Reply by BartoszCichecki** [reply]
Without a log or some sort of details, it's impossible to tell what's wrong.

Configure run step, press Run now and if you see ran successfully, all is working fine.

**Reply by Ace-Radom** [reply]
@Wittusus please try to have a look at the task manager while simply starting the cmd.

![image](https://github.com/BartoszCichecki/LenovoLegionToolkit/assets/104905842/fc8da60c-b28c-4da8-a3fe-45515dfcae0c)

In this case, you should see something like this:

![image](https://github.com/BartoszCichecki/LenovoLegionToolkit/assets/104905842/a99f772c-bc9a-4386-9a2d-f6ac417c1b83)

Just for translation: from top to bottom: LLT itself, cmd, console host process (`conhost.exe`).

As discussed in #1216, it is excepted & designed that no console window is shown when running a full-console program / script. Therefore I don't think your script is not "run", but not "shown“. If you have tried the above steps and see the same thing in the task manager, then the problem is easy to point out. A new feature which should fix your problem is added in #1217 and should be released with `2.21.0`, I think. Please uncheck the `Run Silently` check-box and try again when you get the next release.

But if you tried the above steps and didn't see the subprocesses (cmd, conhost) under LLT, or you saw these things but your scripts just don't work, then without a log we can do nothing. In this case it's impossible for us to reproduce your problem on our devices, without your log file.
